File: benchmarks/sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int sieve(int n) {
    char *flags = (char *)calloc(n + 1, 1);
    int count = 0;
    for (int i = 2; i <= n; i++) {
        if (!flags[i]) {
            count++;
            for (long long k = (long long)i * i; k <= n; k += i) {
                flags[k] = 1;
            }
        }
    }
    free(flags);
    return count;
}

int main(void) {
    printf("%d\n", sieve(100000));
    return 0;
}

File: benchmarks/sieve.py
def sieve(n):
    flags = bytearray(n + 1)
    count = 0
    for i in range(2, n + 1):
        if not flags[i]:
            count += 1
            for k in range(i * i, n + 1, i):
                flags[k] = 1
    return count

print(sieve(100000))
